code/modules/nano/modules/narrate_panel.dm

```diff
@@ -0,0 +1,67 @@
+/datum/tgui_module/narrate_panel/ui_interact(var/mob/user, var/datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "NarratePanel", "Narrate Panel", 800, 400)
+		ui.open()
+
+/datum/tgui_module/narrate_panel/ui_data(mob/user)
+	var/list/data = list()
+	data["narrate_styles"] = list("danger", "notice", "warning", "alien", "cult")
+	data["narrate_locations"] = list("View", "Range", "Z-Level", "Global")
+	return data
+
+/datum/tgui_module/narrate_panel/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+
+	var/mob/abstract/ghost/ghost = ui.user
+	if(!istype(ghost))
+		return FALSE
+
+	if(action == "narrate")
```

Comment:
Maybe we should use defines for this? Eh...

Reply:
no point for a thing only used by one topic call